src/town_cmd.cpp

```diff
@@ -1228,17 +1228,44 @@ static void GrowTownInTile(TileIndex *tile_ptr, RoadBits cur_rb, DiagDirection t
 		/* Possibly extend the road in a direction.
 		 * Randomize a direction and if it has a road, bail out. */
 		target_dir = RandomDiagDir();
-		if (cur_rb & DiagDirToRoadBits(target_dir)) return;
-
-		/* This is the tile we will reach if we extend to this direction. */
-		TileIndex house_tile = TileAddByDiagDir(tile, target_dir); // position of a possible house
+		RoadBits target_rb = DiagDirToRoadBits(target_dir);
+		TileIndex house_tile; // position of a possible house
+
+		if (cur_rb & target_rb) {
+			/* If it's a road turn possibly build a house in a corner.
+			 * Use intersection of turn with straight road (always 1 bit)
+			 * as an indicator that we randomed corner house position */
+			if ((cur_rb & ROAD_X) != target_rb) return;
```

Comment:
Is there some reasoning to randomise this by 50% ?
I assume there are 3 positions for a house at a corner, and the non-corner positions are taken with a chance of 1/4. So the corner shall also have 1/4?
Expand the comment about that.